lib/bolt/pal.rb

```diff
@@ -1,15 +1,27 @@
-# TODO: This is currently used only for testing. I will refactor the CLI to use
-# this in a separate PR
+require 'bolt/executor'
+require 'bolt/error'
+
 module Bolt
   class PAL
     BOLTLIB_PATH = File.join(__FILE__, '../../../modules')
 
     def initialize(config)
-      # TODO: how should we manage state? Does noop go here?
+      # Nothing works without initialized this global state. Reinitializing
+      # is safe and in practice only happen in tests
+      self.class.load_puppet
+      self.class.configure_logging(config[:log_level])
+
+      # TODO: how should we manage executor state?
```

Comment:
Not sure what this means.

Reply:
Why not pass the executor in to run_task/run_plan? Seems as straight-forward as passing `noop` through.